packages/ui/components/SchemaForm.tsx

```diff
@@ -27,37 +27,36 @@ export type SchemaFormProps<TSchema extends z.ZodTypeAny> = Omit<
 }
 
 function titleCase(str: string) {
-  return str
-    .split('_')
-    .map((word) => word.charAt(0).toUpperCase() + word.slice(1))
+  const words = str.split(/(?=[A-Z])|_/)
+
+  return words
+    .map((word) => word.charAt(0).toUpperCase() + word.toLowerCase().slice(1))
     .join(' ')
 }
 
+function isTypeObject(schema: RJSFSchema): boolean {
+  return (
+    schema.type === 'object' ||
+    (Array.isArray(schema.type) && schema.type.includes('object'))
+  )
+}
 // Add this function before the SchemaForm component
 function generateUiSchema(jsonSchema: RJSFSchema): UiSchema {
   const uiSchema: UiSchema = {}
 
-  if (jsonSchema.type === 'object' && jsonSchema.properties) {
+  if (isTypeObject(jsonSchema) && jsonSchema.properties) {
     for (const [key, value] of Object.entries(jsonSchema.properties)) {
       const friendlyLabel = titleCase(key)
       uiSchema[key] = {
         'ui:title': friendlyLabel,
         'ui:classNames': 'pt-2 mr-2',
       }
 
-      if (typeof value === 'object' && value.type === 'object') {
+      if (typeof value === 'object' && isTypeObject(value)) {
```

Comment:
this seems duplicated, why call typeof value === 'object' here?

Reply:
Just for typescript to know that value.type could exist and we can verify it mostly, also value could be Array like in other fields from jsonSchema or a string describing the type of the field.

Reply:
Probably I should rename the boolean func better, but it checks specifically the schema.type is object